Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,512 @@
+        sceneView.isBackgroundTransparent = true
+        sceneView.atmosphereEffect = .none
+        
+        // Tell the sceneView we will be calling `renderFrame()` manually.
+        sceneView.isManualRendering = true
+        
+        // We haven't yet notified user of start or failure.
+        notifiedStartOrFailure = false
+        
+        // Set intitial orientationQuat.
+        orientationChanged(notification: nil)
+    }
+    
+    // MARK: Public
+    
+    /// Determines the map point for the given screen point.
+    ///
+    /// - Parameter toLocation: The point in screen coordinates.
+    /// - Returns: The map point corresponding to screenPoint.
+    public func arScreenToLocation(screenPoint: AGSPoint) -> AGSPoint {
+        fatalError("arScreen(toLocation:) has not been implemented")
+    }
+    
+    /// Resets the device tracking, using `originCamera` if it's not nil or the device's GPS location via the locationManager.
+    public func resetTracking() {
+        initialLocation = nil
+        startTracking()
+    }
+
+    /// Resets the device tracking, using the device's GPS location via the locationManager.
+    ///
+    /// - Returns: Reset operation success or failure.
+    public func resetUsingLocationServices() -> Bool {
+        fatalError("resetUsingLocationServices() has not been implemented")
+    }
+    
+    /// Resets the device tracking using a spacial anchor.
+    ///
+    /// - Returns: Reset operation success or failure.
+    public func resetUsingSpatialAnchor() -> Bool {
+        fatalError("resetUsingSpatialAnchor() has not been implemented")
+    }
+    
+    /// Starts device tracking.
+    public func startTracking() {
+        if !isSupported {
+            didStartOrFailWithError(ARError(.unsupportedConfiguration))
+            return
+        }
+        
+        if let origin = originCamera {
+            // We have a starting camera.
+            initialTransformationMatrix = origin.transformationMatrix
+            sceneView.setViewpointCamera(origin)
+            finalizeStart()
+        }
+        else {
+            // No starting camera, use location manger to get initial location.
+            let authStatus = CLLocationManager.authorizationStatus()
+            switch authStatus {
+            case .notDetermined:
+                startWithAccessNotDetermined()
+            case .restricted, .denied:
+                startWithAccessDenied()
+            case .authorizedAlways, .authorizedWhenInUse:
+                startWithAccessAuthorized()
+            }
+        }
+        
+        //TODO:  reloook at the mechanism by which we're getting notified of orientation changed events...
+        // We need to know when the device orientation changes in order to update the Camera transformation.
+        UIDevice.current.beginGeneratingDeviceOrientationNotifications()
+        NotificationCenter.default.addObserver(
```

Comment:
In some cases, this won't happen until after finalizeStart is called. `finalizeStart` looks like it makes callbacks, which if at that point the user does something to stopTracking, then the observer will get removed from the notification center before it's added. This used to be a crash (not sure if it still is, but I think so), however, at the least what could happen is that the observer is then added after it's removed, so it won't get properly removed.

Reply:
Added a boolean tracking whether the observer was added or not.  Also now using the more specific `removeObserver(_:name:object:)` method per the Apple doc.